src/utils/breadcrumbs.ts

```diff
@@ -0,0 +1,209 @@
+import { NextPageContext } from 'next';
+import {
+    PASSENGER_TYPE_COOKIE,
+    FARE_TYPE_COOKIE,
+    INPUT_METHOD_COOKIE,
+    JOURNEY_COOKIE,
+    PERIOD_TYPE_COOKIE,
+} from '../constants/index';
+import { Breadcrumb } from '../interfaces';
+import { getCookieValue } from '.';
+
+export default (ctx: NextPageContext): { generate: () => Breadcrumb[] } => {
+    const url = ctx.req?.url;
+
+    if (!url || url === '/') {
+        return {
+            generate: (): Breadcrumb[] => [],
+        };
+    }
+
+    const fareType = getCookieValue(ctx, FARE_TYPE_COOKIE, 'fareType');
+    const outboundJourney = getCookieValue(ctx, JOURNEY_COOKIE, 'outboundJourney');
+    const inputMethod = getCookieValue(ctx, INPUT_METHOD_COOKIE, 'inputMethod');
```

Comment:
wanted to ask what the purpose of this cookie is?

Reply:
It's there so we know which input method the user selected to in turn show the correct breadcrumbs. It's used on both ln. 45 & 46.